web_src/js/features/notification.js

```diff
@@ -177,14 +178,15 @@ async function updateNotificationCount() {
 
     const data = await response.json();
 
-    const $notificationCount = $('.notification_count');
     if (data.new === 0) {
-      $notificationCount.addClass('tw-hidden');
+      hideElem('.notification_count');
     } else {
-      $notificationCount.removeClass('tw-hidden');
+      showElem('.notification_count');
     }
 
-    $notificationCount.text(`${data.new}`);
+    for (const el of document.getElementsByClassName('notification_count')) {
+      el.textContent = `${data.new}`;
+    }
```

Comment:
I guess it's better to introduce a util function:

```
queryElem('.notification_count', (el) => el.textContent = `${data.new}`);
```

I have seen too many loops for the refactoring.

Reply:
Maybe export https://github.com/go-gitea/gitea/blob/849eee8db70c8999d54350b85ea7a16fc44dc404/web_src/js/utils/dom.js#L3?

Reply:
Yup, it is a good candidate, meanwhile I guess it needs some rewriting and renaming.

Reply:
[`99d4ea3` (#30172)](https://github.com/go-gitea/gitea/pull/30172/commits/99d4ea3b08f426ea29eefcb1dd07258a66db5c5c)

Feel free to push here for nits

Reply:
I would name it `mapElems` or `forElemens`.

Reply:
Maybe we can land as-is and rename in another PR?

Reply:
I would say revert the changes here and make another PR that introduces the new function and refactor all existing for loops.

Reply:
@wxiaoguang would you still approve if I revert [`99d4ea3` (#30172)](https://github.com/go-gitea/gitea/pull/30172/commits/99d4ea3b08f426ea29eefcb1dd07258a66db5c5c)?

Reply:
I'd say definitely do not use `elementsCall` because that has some baggage on it that we don't need for this new function, which would just be:

```js
export function forElems(selector, cb) {
  for const el of document.querySelectorAll(selector) {
    cb(el);
  }
}
```
Thought honstely I don't have much of a problem with those `for` loops and callback style has problems as well because it's not possible to `continue` or `break` in them. So I guess we may as well just keep the loops.

Reply:
Could also just use built-in methods, a bit more verbose but does the same:

```js
Array.from(document.querySelectorAll('.foo')).forEach((el) => el.checked = true);

Reply:
Looks there is [`NodeList.forEach`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach), so it can be shortened:

```
document.querySelectorAll('.foo').forEach((el) => el.checked = true);
```

@wxiaoguang @yardenshoham is that acceptable to you?

Reply:
> Could also just use built-in methods

Lint converts it into a for-of loop.
![image](https://github.com/go-gitea/gitea/assets/20454870/841e5c78-5e0d-4b28-8cb8-266bb80b94de)

> Looks there is [NodeList.forEach](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach), so it can be shortened

Same here.
![image](https://github.com/go-gitea/gitea/assets/20454870/ac080a02-20d7-42ee-ba6c-d462d0333ce7)

---

Honestly I think using a loop is OK, I don't see a big need to avoid it.

Reply:
Hmm yes the lint rule thinks it's `Array.forEach` because it has no type information and this is unfixable in the rule unless it gains type information from things like typescript's parser which we don't use yet.

So yes, I prefer revert to `for-of` now, even if it's more verbose.

Reply:
https://github.com/go-gitea/gitea/pull/30172/commits/892cdc75a882b890356c0a572679efbf8c167f8e